### ISSUE TITLE
`SimpleDateFormat` is not threadsafe

### ISSUE THREAD

**Issue by bmorris591** [nitram509/jmacaroons#4]
Browsing through the code I found the following in [`TimestampCaveatVerifier`][1]:

```java
private static SimpleDateFormat ISO_DateFormat_DAY = new SimpleDateFormat("yyyy-MM-dd");
private static SimpleDateFormat ISO_DateFormat_HOUR = new SimpleDateFormat("yyyy-MM-dd'T'HH");
private static SimpleDateFormat ISO_DateFormat_MINUTE = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm");
private static SimpleDateFormat ISO_DateFormat_SECOND = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss");
private static SimpleDateFormat ISO_DateFormat_TIMEZONE = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssZ");
```

As these variables are `static`, they are shared between all `TimestampCaveatVerifier` instances.  Therefore use of the `TimestampCaveatVerifier` in a multithreaded environment is impossible, unless one shares a single instance and synchronises it.

Consider making the `SimpleDateFormat` instances non-`static`.  If performance is a concern then a [`BlockingQueue`][2] can be used as a pool.

[1]: https://github.com/nitram509/jmacaroons/blob/master/src/main/java/com/github/nitram509/jmacaroons/verifier/TimestampCaveatVerifier.java
[2]: https://docs.oracle.com/javase/7/docs/api/java/util/concurrent/BlockingQueue.html

**Comment by nitram509** [comment]
Thank you very much for pointing this out.
I've commited a fix in [04b5ba1478507db0264aa3acb8a61177bdb6c938]

**Comment by nitram509** [comment]
Thank you very mich for this contribution.
